### PR TITLE
more fixes on console pages 

### PR DESCRIPTION
**Pull request by kolharsam**
All the detailed updates from this PR are on here: https://www.dropbox.com/scl/fi/fgxeck6wplvidc4u95wsz/Metadata-separation-and-multiple-data-sources.paper?dl=0&oref=e&rlkey=dmf2ntwpfwq8271icdlpr92ea

This PR contains changes for:
- Minor changes in Actions(API calls, displaying the correct info regarding the actions)
- Minor changes in Remote Schemas
- Changes from older PRs that were closed
         - deleteFunctionSql changes
         - added some notes to unsupported features
         - add source dropdown UI
         - Raw SQL pages changes

### CODE REVIEW

File: console/src/components/Common/utils/v1QueryUtils.ts [hunk 1]
```diff
@@ -103,26 +103,17 @@ export const getEnumOptionsQuery = (
...
-  type: 'select',
+  type: 'run_sql',
   args: {
-    table: {
-      name: 'event_triggers',
-      schema: 'hdb_catalog',
-    },
-    columns: ['*'],
-    order_by: [makeOrderBy('name', 'asc')],
+    sql: 'SELECT metadata FROM "hdb_catalog"."hdb_metadata";',
```
Comment (beerose) [inline console/src/components/Common/utils/v1QueryUtils.ts]: Please remove

File: console/src/components/Common/utils/v1QueryUtils.ts [hunk 2]
```diff
@@ -103,26 +103,17 @@ export const getEnumOptionsQuery = (
...
+  type: 'run_sql',
   args: {
-    table: {
-      name: 'hdb_cron_triggers',
-      schema: 'hdb_catalog',
-    },
-    columns: ['*'],
-    order_by: [makeOrderBy('name', 'asc')],
+    sql:
+      'SELECT * FROM "hdb_catalog"."hdb_cron_triggers" ORDER BY name ASC NULLS LAST;',
```
Comment (beerose) [inline console/src/components/Common/utils/v1QueryUtils.ts]: Const names says `scheduled` and you're fetching `cron` triggers.

File: console/src/components/Services/Data/RawSQL/Actions.js [hunk 3]
```diff
@@ -41,13 +41,17 @@ const executeSQL = (isMigration, migrationName, statementTimeout) => (
   const { isTableTrackChecked, isCascadeChecked, sql } = getState().rawSQL;
   const { migrationMode, readOnlyMode } = getState().main;
   const isStatementTimeout = statementTimeout && !isMigration;
-
+  const { currentDataSource } = getState().tables;
```
Comment (beerose) [inline console/src/components/Services/Data/RawSQL/Actions.js]: The same thing was declared a few lines below.

File: console/src/components/Services/Data/RawSQL/Actions.js [hunk 4]
```diff
@@ -41,13 +41,17 @@ const executeSQL = (isMigration, migrationName, statementTimeout) => (
...
   const migrateUrl = returnMigrateUrl(migrationMode);
 
   let url = Endpoints.query;
   const source = getState().tables.currentDataSource;
   const schemaChangesUp = [];
 
+  schemaChangesUp.push(
+    getRunSqlQuery(sql, source, isCascadeChecked, readOnlyMode)
+  );
+
```
Comment (beerose) [inline console/src/components/Services/Data/RawSQL/Actions.js]: This was below on purpose. Statement timeout needs to be executed before a query. Otherwise using it doesn't make sense.

File: console/src/components/Services/Data/DataActions.js [hunk 5]
```diff
@@ -257,9 +257,16 @@ const fetchAdditionalColumnsInfo = () => (dispatch, getState) => {
   return dispatch(requestAction(Endpoints.query, options)).then(
     data => {
       if (data.result) {
+        const makeResult = data.result.slice(1).map(arr => ({
```
Comment (beerose) [inline console/src/components/Services/Data/DataActions.js]: `parseColumnsInfoResult` should take care of this

File: console/src/components/Services/Data/DataActions.js [hunk 6]
```diff
@@ -257,9 +257,16 @@ const fetchAdditionalColumnsInfo = () => (dispatch, getState) => {
...
+          column_name: arr[0],
+          table_name: arr[1],
+          is_generated: arr[2],
+          is_identity: arr[3],
+          identity_generation: arr[4] === 'NULL' ? null : arr[4],
+        }));
         dispatch({
           type: SET_ADDITIONAL_COLUMNS_INFO,
-          data: dataSource.parseColumnsInfoResult(data.result),
+          data: dataSource.parseColumnsInfoResult(makeResult),
```
Comment (kolharsam) [inline console/src/components/Services/Data/DataActions.js]: move to parseColumnsInfoResult

File: console/src/components/Services/Events/EventTriggers/Modify/Modify.tsx [hunk 7]
```diff
@@ -20,14 +20,25 @@ import { mapDispatchToPropsEmpty } from '../../../../Common/utils/reactUtils';
...
-
-  const { state, setState } = useEventTriggerModify(currentTrigger, allSchemas);
+  const {
+    currentTrigger,
+    allSchemas,
+    readOnlyMode,
+    dispatch,
+    metadataObject,
+  } = props;
+  console.log({ currentTrigger, allSchemas });
```
Comment (kolharsam) [inline console/src/components/Services/Events/EventTriggers/Modify/Modify.tsx]: remove

File: console/src/components/Services/Events/EventTriggers/Modify/Modify.tsx [hunk 8]
```diff
@@ -20,14 +20,25 @@ import { mapDispatchToPropsEmpty } from '../../../../Common/utils/reactUtils';
 import { modifyEventTrigger, deleteEventTrigger } from '../../ServerIO';
```
Comment (kolharsam) [inline console/src/components/Services/Events/EventTriggers/Modify/Modify.tsx]: move this event trigger change to another PR

File: console/src/components/Services/Events/EventTriggers/state.ts [hunk 9]
```diff
@@ -16,7 +16,11 @@ import {
 import { parseServerHeaders } from '../../../Common/Headers/utils';
 import { Table } from '../../../../dataSources/types';
 import { generateTableDef } from '../../../../dataSources';
-import { QualifiedTable } from '../../../../metadata/types';
+import {
```
Comment (kolharsam) [inline console/src/components/Services/Events/EventTriggers/state.ts]: move to different PR

File: console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js [hunk 10]
```diff
@@ -189,21 +186,24 @@ const addRemoteSchema = () => {
...
-      // alert(JSON.stringify(err));
     };
     dispatch({ type: ADDING_REMOTE_SCHEMA });
+
     return dispatch(
       makeRequest(
-        upQuery.args,
-        downQuery.args,
+        [upQuery],
+        [downQuery],
```
Comment (kolharsam) [inline console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js]: check this once again

File: console/src/metadata/queryUtils.ts [hunk 11]
```diff
@@ -417,9 +420,11 @@ export const generateCreateEventTriggerQuery = (
     replace,
   });
 
-export const getDropEventTriggerQuery = (name: string) => ({
-  type: 'delete_event_trigger',
+export const getDropEventTriggerQuery = (name: string, source: string) => ({
+  // Not supported for MySQL
+  type: 'pg_delete_event_trigger',
```
Comment (kolharsam) [inline console/src/metadata/queryUtils.ts:425]: remove

**Review by kolharsam — COMMENTED** [review]
Some notes for myself.

File: console/src/components/Services/Data/Function/customFunctionReducer.js [hunk 12]
```diff
@@ -161,69 +161,45 @@ const fetchCustomFunction = (functionName, schema, source) => {
...
+  const successMsg = 'Function deleted';
+  const errorMsg = 'Deleting function failed';
+
+  const customOnSuccess = () =>
+    dispatch(_push(getSchemaBaseRoute(currentSchema)));
+  const customOnError = () => dispatch({ type: DELETE_CUSTOM_FUNCTION_FAIL });
+
+  dispatch({ type: DELETING_CUSTOM_FUNCTION });
+  return dispatch(
+    makeRequest(
```
Comment (kolharsam) [inline console/src/components/Services/Data/Function/customFunctionReducer.js:192]: check

File: console/src/components/Services/Events/EventTriggers/Add/Add.tsx [hunk 13]
```diff
@@ -18,6 +18,8 @@ import * as tooltip from '../Common/Tooltips';
 import { EVENTS_SERVICE_HEADING } from '../../constants';
 import { mapDispatchToPropsEmpty } from '../../../../Common/utils/reactUtils';
 import { Table } from '../../../../../dataSources/types';
+import { getDataSources } from '../../../../../metadata/selector';
```
Comment (kolharsam) [inline console/src/components/Services/Events/EventTriggers/Add/Add.tsx:26]: move these changes to diff. PR

File: console/src/components/Services/Events/EventTriggers/Common/Tooltips.js [hunk 14]
```diff
@@ -3,7 +3,7 @@ import Tooltip from 'react-bootstrap/lib/Tooltip';
 
 export const manualTriggerInfo = (
   <Tooltip id="tooltip-manual-trigger-description">
-    Selecting this option will let you invoke this trigger from Data Broeser
+    Selecting this option will let you invoke this trigger from Data Browser
```
Comment (kolharsam) [inline console/src/components/Services/Events/EventTriggers/Common/Tooltips.js:6]: move to Diff PR

File: console/src/dataSources/services/mysql/index.tsx [hunk 15]
```diff
@@ -139,6 +139,9 @@ export const mysql: DataSourcesAPI = {
   findFunction: () => {
     return undefined;
   },
+  deleteFunctionSql: () => {
+    throw new Error('not supported');
```
Comment (kolharsam) [inline console/src/dataSources/services/mysql/index.tsx]: make `undefined`

**Review by kolharsam — COMMENTED** [review]
more comments that need work

File: console/src/components/Services/Data/RawSQL/Actions.js [hunk 16]
```diff
@@ -126,11 +126,34 @@ const executeSQL = (isMigration, migrationName, statementTimeout) => (
...
+        trackQuery.type = 'add_existing_table_or_view';
+      }
+
+      trackQuery.args.name = object.name;
+      trackQuery.args.schema = object.schema;
+
+      schemaChangesUp.push(trackQuery);
+    });
+  }
+
```
Comment (beerose) [inline console/src/components/Services/Data/RawSQL/Actions.js]:
```suggestion
```

File: console/src/components/Services/Data/TableModify/ModifyActions.js [hunk 17]
```diff
@@ -320,6 +320,7 @@ export const setCustomRootFields = successCb => (dispatch, getState) => {
     dispatch({ type: UPDATE_MIGRATION_STATUS_ERROR, data: err });
   };
 
+  // TODO: this call is to be made on v1/metadata
```
Comment (beerose) [inline console/src/components/Services/Data/TableModify/ModifyActions.js]:
```suggestion
```

File: console/src/components/Services/Data/TableModify/ModifyTable.js [hunk 18]
```diff
@@ -52,6 +52,28 @@ import {
 } from '../Common/TooltipMessages';
 import { RightContainer } from '../../../Common/Layout/RightContainer';
 
+const NotSupportedNote = ({ currentDataSource, unSupportedSources }) => {
```
Comment (beerose) [inline console/src/components/Services/Data/TableModify/ModifyTable.js]: comment for myself: todo

File: console/src/components/Services/Data/TableModify/ModifyTable.js [hunk 19]
```diff
@@ -189,13 +211,19 @@ class ModifyTable extends React.Component {
...
+            <h4 className={styles.subheading_text_no_padd}>
+              Computed fields
+              <Tooltip
+                message={'Add a function as a virtual field in the GraphQL API'}
+              />
+              <KnowMoreLink href="https://hasura.io/docs/1.0/graphql/manual/schema/computed-fields.html" />
+            </h4>
+            <NotSupportedNote
+              currentDataSource={currentSource}
+              unSupportedTypes={['mysql']}
```
Comment (beerose) [inline console/src/components/Services/Data/TableModify/ModifyTable.js:224]: ditto

File: console/src/components/Services/Events/EventTriggers/Common/Tooltips.js [hunk 20]
```diff
@@ -49,3 +49,9 @@ export const statusCodeDescription = (
     Status code of the webhook response
   </Tooltip>
 );
+
+export const dataSourceDescription = (
+  <Tooltip id="tooltip-trigger-status-code-description">
+    Select the data source where the table for the data trigger exists
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/Common/Tooltips.js:55]: comment for myself: todo

File: console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js [hunk 21]
```diff
@@ -189,17 +186,20 @@ const addRemoteSchema = () => {
         dispatch({ type: RESET }),
         dispatch(exportMetadata()).then(() => {
           dispatch(push(`${prefixUrl}/manage/${resolveObj.name}/details`));
+          dispatch({
+            type: SET_REMOTE_SCHEMAS,
+            data: getState().metadata.metadataObject.remote_schemas,
+          });
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js]:
```suggestion
```

File: console/src/components/Services/RemoteSchema/RemoteSchemaRouter.js [hunk 22]
```diff
@@ -12,7 +12,6 @@ import {
 import { FILTER_REMOTE_SCHEMAS } from './Actions';
 
 import { appPrefix } from './constants';
-import { getRemoteSchemasSelector } from '../../../metadata/selector';
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/RemoteSchemaRouter.js:15]: bring back selector

File: console/src/dataSources/services/mysql/index.tsx [hunk 23]
```diff
@@ -139,6 +138,7 @@ export const mysql: DataSourcesAPI = {
   findFunction: () => {
     return undefined;
   },
+  deleteFunctionSql: () => undefined,
```
Comment (beerose) [inline console/src/dataSources/services/mysql/index.tsx]:
```suggestion
  deleteFunctionSql: undefined,
```

File: console/src/dataSources/index.ts [hunk 24]
```diff
@@ -276,6 +276,10 @@ export interface DataSourcesAPI {
   }) => string;
   getFKRelations: (options: { schemas: string[]; tables: Table[] }) => string;
   getReferenceOption: (opt: string) => string;
+  deleteFunctionSql: (
+    schemaName: string,
+    functionState: FunctionState
+  ) => string | undefined;
```
Comment (beerose) [inline console/src/dataSources/index.ts]:
```suggestion
  deleteFunctionSql?: (
    schemaName: string,
    functionState: FunctionState
  ) => string;
```

File: console/src/metadata/selector.ts [hunk 25]
```diff
@@ -17,19 +17,27 @@ const isMetadataV3 = (
...
+      source => source.name === currentDataSource
+      // NOTE: Commented this since, kind is not being mentioned on the metadata object atm
+      // &&
+      // (source.kind || 'postgres') === currentDriver
     );
   }
   return state.metadata.metadataObject;
 };
 
+export const getRemoteSchemasFromMetadata = (state: ReduxState) => {
```
Comment (beerose) [inline console/src/metadata/selector.ts]:
```suggestion
export const getRemoteSchemas = (state: ReduxState) => {
```

File: console/src/components/Services/Events/EventTriggers/Add/Add.tsx [hunk 26]
```diff
@@ -71,6 +90,24 @@ const Add: React.FC<Props> = props => {
     setState.table(undefined, selectedSchemaName);
   };
 
+  const handleDataSourceChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedSourceName = e.target.value;
+    // NOTE: I'm not sure if we should be doing this?
+    dispatch({
+      type: UPDATE_CURRENT_DATA_SOURCE,
+      source: selectedSourceName,
+    });
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/Add/Add.tsx]:
```suggestion
```

File: console/src/components/Services/Events/EventTriggers/Add/Add.tsx [hunk 27]
```diff
@@ -195,6 +227,35 @@ const Add: React.FC<Props> = props => {
...
+                Data Source &nbsp; &nbsp;
+                <OverlayTrigger
+                  placement="right"
+                  overlay={tooltip.dataSourceDescription}
+                >
+                  <i className="fa fa-question-circle" aria-hidden="true" />
+                </OverlayTrigger>{' '}
+              </h4>
+              <small>
+                <i>Note: This feature is currently not supported for MySQL</i>
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/Add/Add.tsx:241]: use notsupportednote component

File: console/src/components/Services/Events/EventTriggers/state.ts [hunk 28]
```diff
@@ -71,6 +74,7 @@ export const parseServerETDefinition = (
 
   return {
     name: eventTrigger.name,
+    source: source ?? 'default',
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/state.ts]:
```suggestion
    source: source ?? '',
```

**Review by beerose — APPROVED** [review]
